### PR TITLE
chore: migrate E2E buyEth tests to Foundry

### LINKED ISSUE #135: tests(e2e): `buyEth`

### PR DESCRIPTION
**Pull request by meetmangukiya**
## Description

Migrate buyEth e2e test to foundry. Depends on #215 

## Test Plan

CI

## Related Issues

Closes #135

### CODE REVIEW

**Comment by meetmangukiya** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by meetmangukiya** [comment]
recheck

File: test/e2e/BuyEth.t.sol [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+            domainSeparator,
+            GPv2Signing.Scheme.Eip712,
+            0
+        );
+
+        // give some usdt to trader2
+        _mintUsdt(trader2.addr, 1201.2e6);
+        // approve usdt for trading on the vault
+        vm.startPrank(trader2.addr);
+        USDT.approve(vaultRelayer, 0);
```
Comment (mfw78) [inline test/e2e/BuyEth.t.sol:75]: Is this required? `trader2` shouldn't have any allowance already set for `vaultRelayer` on `USDT`, and therefore it shouldn't need to be zeroed again?

File: test/e2e/BuyEth.t.sol [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            vm,
+            trader2,
+            GPv2Order.Data({
+                sellToken: IERC20(address(USDT)),
+                buyToken: IERC20(GPv2Transfer.BUY_ETH_ADDRESS),
+                receiver: trader2.addr,
+                sellAmount: 1200e6,
+                buyAmount: 1 ether,
+                validTo: 0xffffffff,
+                appData: bytes32(uint256(1)),
```
Comment (mfw78) [inline test/e2e/BuyEth.t.sol:89]: Just confirming it's intentional to leave this set to 1 (`appData` is irrelevant here).

**Review by mfw78 — COMMENTED** [review]
Can this please be rebased on #215 so that CI/CD can pass?

File: test/e2e/BuyEth.t.sol (hunk 2 above)
Comment (meetmangukiya) [reply test/e2e/BuyEth.t.sol:89]: yeah?

File: test/e2e/BuyEth.t.sol (hunk 1 above)
Comment (meetmangukiya) [reply test/e2e/BuyEth.t.sol:75]: yeah was just me being particular about usdt

**Review by mfw78 — APPROVED** [review]

**Pull request closed without merging**